# HG changeset patch
# User Rik <rik@octave.org>
# Date 1346811846 25200
#      Tue Sep 04 19:24:06 2012 -0700
# Node ID f19d0bce85da769df6f1e941f8ef6dc0c08b8fc8
# Parent  43e484fa7e424d32ea1bb9ebffa83f248c40aff7
build: Add check for Qt libraries to configure.ac.

* configure.ac: Add check for Qt libraries.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2184,37 +2184,42 @@ QT_INCDIR=
 QT_LIBDIR=
 build_gui=yes
 win32_terminal=no
 AC_ARG_ENABLE(gui,
   [AS_HELP_STRING([--disable-gui], [don't build the GUI])],
   [if test "$enableval" = no; then build_gui=no; fi], [build_gui=yes])
 
 if test $build_gui = yes; then
+  ## Check for Qt libraries
+  PKG_CHECK_MODULES(QT, [QtCore, QtGui, QtNetwork],
+    [],
+    [AC_MSG_ERROR([Qt libraries are required to build the GUI])])
+
+  ## Check for Qt utility programs
   AC_CHECK_PROGS(QMAKE, [qmake qmake-qt4 qmake-qt5])
   if test -z "$QMAKE"; then
     AC_MSG_ERROR([qmake is required to build the GUI])
   fi
 
   QT_INCDIR=`$QMAKE -query | sed -n -e 's/^QT_INSTALL_HEADERS://p' | sed -e 's,\\\\,/,g'`
   QT_LIBDIR=`$QMAKE -query | sed -n -e 's/^QT_INSTALL_LIBS://p' | sed -e 's,\\\\,/,g'`
 
-  ## FIXME: Need an actual check for Qt libraries.
-  ##        The test below only checks for certain utilities also needed to build the GUI.
   AC_CHECK_PROGS(MOC, [moc moc-qt4 moc-qt5])
   AC_CHECK_PROGS(UIC, [uic uic-qt4 uic-qt5])
   AC_CHECK_PROGS(RCC, [rcc])
-  if test -n "$MOC" && test -n "$UIC" && test -n "$RCC"; then
-    HAVE_QT=true
-    AC_DEFINE(HAVE_QT, 1, 
-      [Define to 1 if Qt is available (must have moc, uic, and rcc programs and developer header files and libraries installed).])
-  fi
+  if test -z "$MOC" || test -z "$UIC" || test -z "$RCC"; then
+    AC_MSG_ERROR([Qt utility programs moc, uic, and rcc are required to build the GUI])
+  fi 
+
+  HAVE_QT=true
+  AC_DEFINE(HAVE_QT, 1, 
+    [Define to 1 if Qt is available (libraries, developer header files, utility programs (qmake, moc, uic, and rcc))])
 
   ## Check for Qscintilla library which is used in the GUI editor. 
-  ## This indirectly tests for Qt libraries
   AC_CACHE_CHECK([whether Qscintilla library is installed],
     [octave_cv_lib_qscintilla],
     [save_CPPFLAGS="$CPPFLAGS"
     save_LDFLAGS="$LDFLAGS"
     save_LIBS="$LIBS"
     CPPFLAGS="-I$QT_INCDIR -I$QT_INCDIR/Qt $CPPFLAGS"
     LDFLAGS="-L$QT_LIBDIR $LDFLAGS"
     LIBS="-lqscintilla2"
